src/sentry/tasks/post_process.py

```diff
@@ -655,17 +655,31 @@ def process_snoozes(job: PostProcessJob) -> None:
     if job["is_reprocessed"] or not job["has_reappeared"]:
         return
 
+    from sentry.issues.escalating import is_escalating
     from sentry.models import (
         Activity,
         GroupInboxReason,
         GroupSnooze,
         GroupStatus,
+        GroupSubStatus,
         add_group_to_inbox,
     )
     from sentry.models.grouphistory import GroupHistoryStatus, record_group_history
 
     group = job["event"].group
 
+    # Check is group is escalating
+    if (
+        features.has("organizations:escalating-issues", group.organization)
+        and group.status == GroupStatus.IGNORED
+        and group.substatus == GroupSubStatus.UNTIL_ESCALATING
+    ):
+        if not is_escalating(group):
+            job["has_reappeared"] = True
+            return
+        job["has_reappeared"] = False
+        return
+
     with metrics.timer("post_process.process_snoozes.duration"):
```

Comment:
It would be great if we could refactor this block into a new function in a different PR.